src/uproot/behaviors/TH1.py

```diff
@@ -109,7 +109,8 @@ def weighted(self):
         """
         True if the histogram has weights (``fSumw2``); False otherwise.
         """
-        raise NotImplementedError(repr(self))
+        sumw2 = self.member("fSumw2")
+        return len(sumw2) > 0 and len(sumw2) == self.member("fNcells")
```

Comment:
Elsewhere, I have been assuming that `fSumw2` can be `nullptr`, probably because it's possible for some ROOT files (some versions of ROOT?).

```suggestion
        return sumw2 is not None and len(sumw2) == self.member("fNcells")
```

Reply:
I would also add a check for `len(sumw2) > 0` to be considered weighted. Probably it makes no difference since having `fNcells` being zero would mean an empty histogram but it should not be considered weighted imho.